drivers/usbhid-ups.c

```diff
@@ -1422,8 +1432,22 @@ static void ups_status_set(void)
 		dstate_delinfo("input.transfer.reason");
 	}
 
-	if (ups_status & STATUS(ONLINE)) {
-		status_set("OL");		/* on line */
+
+	if ((ups_status & STATUS(ONLINE)) && (ups_status & STATUS(DISCHRG))) {
```

Comment:
Given that below you check for part of the same condition, maybe fuse them together?
````
if (ups_status & STATUS(ONLINE)) {
    if (ups_status & STATUS(DISCHRG)) {
        ... lots of new logic about quirk, calibration, message ...
    } else {
        status_set("OL");
    }
} else {
    status_set("OB");
}
````

Reply:
fixed